libs/datalayout/src/main/java/com/linkedin/openhouse/datalayout/generator/OpenHouseDataLayoutStrategyGenerator.java

```diff
@@ -64,7 +75,34 @@ private Optional<DataLayoutStrategy> generateCompactionStrategy() {
     // Retrieve file sizes of all data files.
     Dataset<Long> fileSizes =
         tableFileStats.get().map((MapFunction<FileStat, Long>) FileStat::getSize, Encoders.LONG());
+    return buildDataLayoutStrategy(fileSizes, 1, null);
+  }
+
+  private List<DataLayoutStrategy> generatePartitionLevelCompactionStrategy() {
+    List<DataLayoutStrategy> strategies = new ArrayList<>();
+    List<PartitionStat> partitionStatsList = tablePartitionStats.get().collectAsList();
+    Dataset<FileStat> fileSizesDataset = tableFileStats.get();
+    // For each partition, generate a compaction strategy
+    partitionStatsList.forEach(
+        partitionStat -> {
+          String partitionValue = String.join(", ", partitionStat.getValues());
+          Dataset<Long> fileSizes =
+              fileSizesDataset
+                  .filter(
+                      (FilterFunction<FileStat>)
+                          fileStat ->
+                              String.join(", ", fileStat.getPartitionValues())
+                                  .equals(partitionValue))
+                  .map((MapFunction<FileStat, Long>) FileStat::getSize, Encoders.LONG());
+          Optional<DataLayoutStrategy> strategy =
+              buildDataLayoutStrategy(fileSizes, partitionStatsList.size(), partitionValue);
+          strategy.ifPresent(strategies::add);
+        });
+    return strategies;
+  }
 
+  private Optional<DataLayoutStrategy> buildDataLayoutStrategy(
+      Dataset<Long> fileSizes, int partitionCount, String partitionValue) {
```

Comment:
Is it partitionValue or partitionValues, if there are more than one partition column?

Reply:
partitionValue is the string representation of list of partitionValues.